shared/constants/permissions.ts

```diff
@@ -31,8 +31,43 @@ export const RestrictedMethods = Object.freeze({
 export const ConnectionPermission = Object.freeze({
   connection_permission: 'connection_permission',
 });
+
+// This configuration specifies permission weight thresholds used to determine which
+// permissions to show or hide on certain Snap-related flows (Install, Update, etc.)
+export const PermissionWeightThreshold = Object.freeze({
+  snapInstall: 3 as const,
+  snapUpdateApprovedPermissions: 2 as const,
+});
 ///: END:ONLY_INCLUDE_IF
 
+export const PermissionWeight = Object.freeze({
+  eth_accounts: 3,
+  permittedChains: 3,
+  snap_dialog: 4,
+  snap_notify: 4,
+  snap_getBip32PublicKey: 3,
+  snap_getBip32Entropy: 1,
+  snap_getBip44Entropy: 1,
+  snap_getEntropy: 3,
+  snap_manageState: 4,
+  snap_getLocale: 4,
+  wallet_snap: 4,
+  endowment_networkAccess: 3,
```

Comment:
```suggestion
  endowment:network-access: 3,
```

Can we use the proper formatting here? Or do you think it gets too complex?

Reply:
It somehow looks bad. It's more difficult to access and error prone while writing code. Because there is no way to define that notation without defining it as the string with quotes, then there is no way to access it directly without having to do something like `weight: PermissionWeight['endowment:networkAccess'],`. 

When these keys are defined like `endowment_networkAccess: 3` it's easier to use it in this way: `weight: PermissionWeight.endowment_networkAccess,` with bunch of other safety mechanisms provided by any IDE.

IMO it's easier to handle and maintain. The issue is the notation we have with `:` and `-`. If you think there are strong reasons to make this the other way for consistency or else, I can make a change.

Reply:
Fair enough 👍 Let's keep it as-is